scripts/pxc-package-prechecks.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash
+set -e
+
+compare_versions() {
+    local version1="$1"
+    local version2="$2"
+
+    if [[ "$version1" == "$version2" ]]; then
+        result=0
+    fi
+
+    local IFS='.'
+    local version1_parts=($version1)
+    local version2_parts=($version2)
+
+    # Compare each part of the version numbers
+    local max_parts=$(( ${#version1_parts[@]} > ${#version2_parts[@]} ? ${#version1_parts[@]} : ${#version2_parts[@]} ))
+    for ((i = 0; i < max_parts; i++)); do
+        local part1="${version1_parts[i]:-0}"
+        local part2="${version2_parts[i]:-0}"
+
+        if (( part1 > part2 )); then
+            result=1
+        elif (( part1 < part2 )); then
+            result=2
+        fi
+    done
+
+
+if [ "$result" -eq 0 ] || [ "$result" -eq 1 ]; then
+    return 1
+else
+    return 0
+fi
+}
+
+
+
+
+
+
+if [ "$#" -ne 1 ]; then
+  echo "This script requires product parameter: ps56, ps57, ps80, ps81 or PS82 !"
+  echo "Usage: ./package_check.sh <prod>"
+  exit 1
+fi
+
+SCRIPT_PWD=$(cd `dirname $0` && pwd)
+
+function check_xb_file(){
+        if [[ -f /usr/bin/pxc_extra/pxb-$1/bin/xtrabackup ]]; then
+            return 1
+        else
+            return 0
+        fi
+}
+
+
+if [[ $1 =~ ^pxc8[1-9]{1}$ ]]; then
+  echo "Proper PXC VERSION"
+
+else
+  echo "Illegal product selected!"
+  exit 1
+fi
+
+
+if [[ $1 =~ ^pxc8[0-9]{1}$ ]]; then
+  if [ -f /usr/bin/wsrep_sst_xtrabackup-v2 ]; then
+    echo "File exists: /usr/bin/wsrep_sst_xtrabackup-v2"
+
+    pxc_version=$(mysql --version  | grep Ver | awk '{print$3}')
```

Comment:
mysqld 

mysql is client 'binary'. The versions should be the same but let's use the mysqld